keystore/src/migrations.rs

```diff
@@ -180,3 +221,40 @@ pub(crate) fn make_ciphersuite_for_signature_scheme(
     };
     Ok(ciphersuite_for_signature_scheme)
 }
+
+pub(crate) fn make_least_used_ciphersuite(
+    persisted_mls_groups: impl IntoIterator<Item = PersistedMlsGroup>,
+) -> CryptoKeystoreResult<impl Fn(u16, u16) -> LeastUsedCiphersuite> {
+    let occurences = count_ciphersuite_occurences(persisted_mls_groups)?;
+
+    let more_likely_ciphersuite = move |ciphersuite_a: u16, ciphersuite_b: u16| -> LeastUsedCiphersuite {
+        let occurence_a = occurences.of(ciphersuite_a);
+        let occurence_b = occurences.of(ciphersuite_b);
+
+        match (occurence_a, occurence_b) {
+            // If one of the occurences is None, it means that the ciphersuites aren't both instances of an ambiguous
+            // pair of ciphersuites.
+            (None, _) => return LeastUsedCiphersuite::Failed,
+            (_, None) => return LeastUsedCiphersuite::Failed,
+            (Some(0), Some(0)) => return LeastUsedCiphersuite::Failed,
+            (Some(a), Some(b)) if a < b => {
+                return LeastUsedCiphersuite::Succeeded(ciphersuite_a);
+            }
+            // If both credentials have equal occurence, let the below if-clause handle this.
+            (Some(a), Some(b)) if a == b => {}
+            _ => {
+                return LeastUsedCiphersuite::Succeeded(ciphersuite_b);
+            }
+        }
+
+        // This is reached when both credentials have equal occurence. Take the one with the numerically
+        // higher ciphersuite in this case.
```

Comment:
Even for internal functions like this, it's nice to put this kind of semantic note into the rustdocs, because docs just show up in modern editors.